content/dotnet-template-azure-iot-edge-module/CSharp/ModuleBackgroundService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+using System.Text;
+
+namespace SampleModule;
+
+internal class ModuleBackgroundService : BackgroundService
+{
+    private int _counter;
+    private ModuleClient? _moduleClient;
+    private readonly ILogger<ModuleBackgroundService> _logger;
+
+    public ModuleBackgroundService(ILogger<ModuleBackgroundService> logger) => _logger = logger;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        MqttTransportSettings mqttSetting = new(TransportType.Mqtt_Tcp_Only);
+        ITransportSettings[] settings = { mqttSetting };
+
+        // Open a connection to the Edge runtime
+        _moduleClient = await ModuleClient.CreateFromEnvironmentAsync(settings);
```

Comment:
Shall we subscribe to the connection callback? I'd at least print out what is happening, but better yet would be to manage client lifetime and be able to close/dispose and reinitialize.

You can look at our [reconnection sample](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples/how%20to%20guides/DeviceReconnectionSample) for patterns to follow here.

Reply:
keep in mind this code will run in a docker container along with $edgeHub, I think there is no need to handle reconnection in this case.

Reply:
Docker containers can never have connection issues to the host?

Reply:
I've not seen any, we could ask the Edge team. Worst case, the process will crash and docker will restart the image.